subconsciousnetwork/subconscious
# Render transclude blocks in MemoDetailViewer

Related to #328 
Resolves #324 
Fixes https://github.com/subconsciousnetwork/subconscious/issues/710

This approach is based on my experiments with rendering in the editor: #428 

# Changes
- Detect slashlinks in subtext blocks
- Fetch transclude previews from the sphere index (SQLite)
- Display inline between blocks
- Translate addresses to match the context
- Tests for fetching local transcludes
    - Difficult to construct a test for 3P content with transcludes

## Own content, own slashlink

<img width="372" alt="image" src="https://github.com/subconsciousnetwork/subconscious/assets/5009316/5d7fae72-f25f-4ed7-b9b2-78635d34d59a">

## 3P Content, their own slashlink

<img width="373" alt="image" src="https://github.com/subconsciousnetwork/subconscious/assets/5009316/78dd7d21-a739-4074-8c18-337f8049f259">


# TODO

- [x] Tests for TranscludeService
    - Insert some notes
    - Force indexing them
    - Call `.fetchTranscludes` with expected links
    - Try different owner values?
- [x] Use `.findAndPushDetail`
